src/main/java/com/kodilla/ecommercee/service/UserService.java

```diff
@@ -0,0 +1,32 @@
+package com.kodilla.ecommercee.service;
+
+import com.kodilla.ecommercee.domain.User;
+import com.kodilla.ecommercee.repository.UserRepository;
+
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@AllArgsConstructor
+@Service
+public class UserService {
+
+    @Autowired
+    private final UserRepository userRepository;
+
+    public User saveUser(final User user) {
+        return userRepository.save(user);
+    }
+
+    public User blockUser(final Long id) {
+        User user = userRepository.findById(id).get();
+        user.setBlocked(true);
+        return userRepository.save(user);
+    }
+
+    public String createUserKey(final Long id) {
+        String userKey = "RANDOM USER KEY OF USER " + id;
+
+        return userKey;
+    }
```

Comment:
wygenerowanie losowego klucza ważnego godzinę po podaniu poprawnych danych użytkownika...

Reply:
wolałbym jakąś podpowiedź tutaj, bo nie mam pomysłu :P

Reply:
Klucz generowany na podstawie Id, hash code, dwa losowe inty a na końcu timestamp. Sprawdzanie ważności moim zdaniem nie musi (albo nawet nie powinno) być w kontrolerze, ponieważ nie przechowujemy klucza w bazie danych. Zapisywany byłby pewnie w ciasteczkach czy coś takiego, a dzięki timestamp'owi łatwo będzie sprawdzić czy minęła godzina czy potrzebny jest nowy klucz :)

Reply:
Do generowania klucza można wykorzystać KeyGeneretor https://www.tutorialspoint.com/java_cryptography/java_cryptography_keygenerator.htm

Co do czasu użyłbym chyba `System.currentTimeMillis();` - jeszcze prościej określić czy minęła godzina 

Ale to tylko sugestie 🙂

Reply:
Słusznie, zawsze lepiej używać gotowych bibliotek. Rano miałam ograniczony czas i zrealizowałem pierwszy pomysł jaki wpadł mi do głowy.